packages/chevrotain/src/parse/grammar/interpreter.ts

```diff
@@ -383,13 +379,13 @@ export function nextPossibleTokensAfter(
   })
 
   while (!isEmpty(possiblePaths)) {
-    const currPath = possiblePaths.pop()
+    const currPath = possiblePaths.pop()!
```

Comment:
looks like isEmpty by lodash could have a type guard for non empty array.
but because the `!` appears just one line under the `isEmpty` call its clear enough...

Reply:
Could also do `while (currPath = possiblePaths.pop())` to avoid a type assertion.

Reply:
Naa, its good, I finally wrapped my head around it :)